src/test/expression/expression_evaluator_test.cpp

```diff
@@ -482,6 +487,45 @@ TEST_F(ExpressionEvaluatorTest, InSelectUncorrelated) {
   EXPECT_TRUE(test_expression<int32_t>(table_a, *in_(34.5, select_b), {std::nullopt}));
   EXPECT_TRUE(test_expression<int32_t>(table_a, *in_("hello", select_b), {0}));
   EXPECT_TRUE(test_expression<int32_t>(table_a, *in_(c, select_b), {1, std::nullopt, 1, std::nullopt}));
+
+  // Test it with pre-calculated uncorrelated_select_results
+  auto uncorrelated_select_results = std::make_shared<ExpressionEvaluator::UncorrelatedSelectResults>();
+  table_wrapper_a->execute();
+  pqp_a->execute();
+  table_wrapper_b->execute();
+  pqp_b->execute();
+  uncorrelated_select_results->emplace(pqp_a, pqp_a->get_output());
+  uncorrelated_select_results->emplace(pqp_b, pqp_b->get_output());
+
+  // Destroy the pqp's so that if the expression evaluator tries to access them, it crashes
+  pqp_a->input_left() = nullptr;
```

Comment:
input_left() returns a value, so this doesn't do what you think it does. And I don't think you intention is a good idea in the first place, since I would be surprised if nullptr-dereferencing is actually guaranteed to crash and not, in fact, undefined behavior.

Reply:
removed it